src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -557,38 +556,131 @@ internal static class HashAlgorithm
         {
             private const string TypeName = "System.Security.Cryptography.HashAlgorithm";
 
-            internal static readonly Type Type = ReflectionUtilities.GetTypeFromEither(
+            private static Type s_lazyType;
+
+            internal static Type Type => s_lazyType ?? (s_lazyType = ReflectionUtilities.GetTypeFromEither(
```

Comment:
What happens if the type is not found? Do we end up with null value and will be repeating lookup on every access of this property? If that is the case, we should use a sentinel value rather than null to indicate uninitialized state. Perhaps typeof(HashAlgorithm).